lib/plugins/aws/deploy/lib/uploadCloudFormationTemplate.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+
+module.exports = {
+  uploadCloudFormationTemplateToS3Bucket() {
+    this.serverless.cli.log('Uploading CloudFormation file to S3...');
+
+    const body = JSON.stringify(this.serverless.service.provider.compiledCloudFormationTemplate);
+
+    const params = {
+      Bucket: this.bucketName,
+      Key: 'compiled-cloudformation-template.json',
+      Body: body,
+    };
+
+    return this.sdk.request('S3',
+      'putObject',
+      params,
+      this.options.stage,
+      this.options.region);
+  },
+
+  uploadCloudFormationTemplate() {
```

Comment:
Just created a new issue for that. Will do the update @eahefnawy mentioned, merge this one and work on the new deployment process as described above / in https://github.com/serverless/serverless/issues/1898